app/database/group_item_store_compute_all_access.go

```diff
@@ -39,35 +53,7 @@ func (s *GroupItemStore) computeAllAccess() {
 		FROM items_items
 		JOIN groups_items AS parents
 			ON parents.item_id = items_items.parent_item_id
-		JOIN groups_items_propagate AS parents_propagate
-			ON parents.id = parents_propagate.id AND parents_propagate.propagate_access = 'children'`
-	stmtInsertMissingChildren, err = s.db.CommonDB().Prepare(queryInsertMissingChildren)
-	mustNotBeError(err)
-	defer func() { mustNotBeError(stmtInsertMissingChildren.Close()) }()
-
-	// inserting missing (or set propagate_access='self' to existing) groups_items_propagate
-	// for groups_items having propagate_access='self'
-	const queryInsertMissingPropagate = `
-		INSERT INTO groups_items_propagate (id, propagate_access)
-		SELECT
-			groups_items.id,
-			'self' as propagate_access
-		FROM groups_items
-		WHERE propagate_access='self'
-		ON DUPLICATE KEY UPDATE propagate_access='self'`
-	stmtInsertMissingPropagate, err = s.db.CommonDB().Prepare(queryInsertMissingPropagate)
-	mustNotBeError(err)
-	defer func() { mustNotBeError(stmtInsertMissingPropagate.Close()) }()
-
-	// Set groups_items as set up for propagation
-	// (switch groups_items.propagate_access from 'self' to 'done')
-	const queryUpdatePropagateAccess = `
-		UPDATE groups_items
-		SET propagate_access='done'
-		WHERE propagate_access='self'`
-	stmtUpdatePropagateAccess, err = s.db.CommonDB().Prepare(queryUpdatePropagateAccess)
-	mustNotBeError(err)
-	defer func() { mustNotBeError(stmtUpdatePropagateAccess.Close()) }()
+		JOIN parents_propagate ON parents_propagate.id = parents.id`
```

Comment:
> I suggest an easy reorganisation in this function:
> I suppose you do not want to extract requests out of the function as it would not make things easier.
> If so, I would suggest to only declare statement without executing any of them in the first part of the function... then draw clearly a line a start the processing, and run the statements that has to be run before the `for hasChanges` loop.
> It will make the "workflow" much clearer.
> 
> BTW, are you sure these requests has necessarily to be run twice (as they are run before the loop and then again as `hasChanged` is true at first.

I don't get what you want to change. We only run queries inside the loop, there are no queries executed before the loop.

Reply:
added more comments